src/pages/AddPurchase/AddPurchase.js

```diff
@@ -0,0 +1,133 @@
+import FoodList from '../../components/FoodList/FoodList'
+import SearchFood from '../../components/SearchFood/SearchFood'
+import styled from 'styled-components/macro'
+import { useState, useEffect } from 'react'
+import getFood from '../../services/getFood'
+import SumFootprint from '../../components/SumFootprint/SumFootprint'
+import PurchaseList from '../../components/PurchaseList/PurchaseList'
+import loadLocally from '../../lib/loadLocally'
+import saveLocally from '../../lib/savelocally'
+
+export default function AddPurchase() {
+  const [foodList, setFoodList] = useState([])
+  const [searchInput, setSearchInput] = useState('')
+  const [purchasedFood, setPurchasedFood] = useState(
+    loadLocally('purchasedFood') ?? []
+  )
+  const [foodListModal, setFoodListModal] = useState(false)
+  const [carbonFootprintSum, setCarbonFootprintSum] = useState(
+    loadLocally('carbonFootprintSum') ?? 0
+  )
+  const [pointerPosition, setPointerPosition] = useState(0)
+
+  useEffect(() => {
+    getFood()
+      .then((data) => setFoodList(data))
+      .catch((error) => console.log(error))
+  }, [])
+
+  function handleChange(event) {
+    setSearchInput(event.target.value)
+  }
+
+  function toggleFoodListModal() {
+    setFoodListModal(!foodListModal)
+  }
+
+  const filteredFoodList = foodList.filter((foodListItem) =>
+    foodListItem.food.toLowerCase().includes(searchInput.toLowerCase())
+  )
+
+  function handleAddFood(id) {
+    const addedItem = foodList.find((foodItem) => foodItem.id === id)
+    setPurchasedFood([...purchasedFood, addedItem])
+    setCarbonFootprintSum(
+      (roundNumber(carbonFootprintSum * 100) +
+        roundNumber(addedItem.co2 * 100)) /
+        100
+    )
+    setFoodListModal(!foodListModal)
+  }
+
+  function handleRemoveFood(index) {
+    const deletedItem = purchasedFood.find(
+      (_, itemPosition) => itemPosition === index
+    )
+    setPurchasedFood(
+      purchasedFood.filter((_, itemPosition) => itemPosition !== index)
+    )
+    setCarbonFootprintSum(
+      (roundNumber(carbonFootprintSum * 100) -
+        roundNumber(deletedItem.co2 * 100)) /
+        100
+    )
+  }
+
+  function roundNumber(number) {
+    return parseFloat(number.toFixed(1))
+  }
+
+  useEffect(() => {
```

Comment:
Please move the useEffects to the top of the component next to the useEffect there. All side effects should be placed together so that it easily to see them